### PR TITLE
Updated Google reCaptcha V1 to Google reCaptcha V2

### PR DESCRIPTION
**Pull request by Lavenblade**
I've updated Google reCaptcha V1 to Google reCaptcha V2. I also updated recaptchalib.php to the latest version: https://github.com/google/recaptcha/blob/1.0.0/php/recaptchalib.php

As you can see, I kinda messed up and ended up with a ton of commits. Sorry about that.

### CODE REVIEW

File: lib/Flux/LoginServer.php [hunk 1]
```diff
@@ -150,22 +150,24 @@ public function register($username, $password, $confirmPassword, $email, $gender
...
 				require_once 'recaptcha/recaptchalib.php';
-				$resp = recaptcha_check_answer(
-					Flux::config('ReCaptchaPrivateKey'),
-					$_SERVER['REMOTE_ADDR'],
-					// Checks POST fields.
-					$_POST['recaptcha_challenge_field'],
-					$_POST['recaptcha_response_field']);
-				
-				if (!$resp->is_valid) {
+				$response;
```
Comment (MishimaHaruna) [inline lib/Flux/LoginServer.php]: Does this line have a purpose?

File: lib/Flux/SessionData.php [hunk 2]
```diff
@@ -263,43 +263,45 @@ public function login($server, $username, $password, $securityCode = null)
...
 				require_once 'recaptcha/recaptchalib.php';
-				$resp = recaptcha_check_answer(
-					Flux::config('ReCaptchaPrivateKey'),
-					$_SERVER['REMOTE_ADDR'],
-					// Checks POST fields.
-					$_POST['recaptcha_challenge_field'],
-					$_POST['recaptcha_response_field']);
-				
-				if (!$resp->is_valid) {
+				$response;
```
Comment (MishimaHaruna) [inline lib/Flux/SessionData.php]: Same as above, does this have a purpose, or is it intended to be `$response = false;` or something similar?

File: lib/Flux/LoginServer.php (hunk 1 above)
Comment (Lavenblade) [inline lib/Flux/LoginServer.php]: It's intended to be $response = false; basically.

I suppose I could change it to $response = $_POST["g-recaptcha-response"]; which is the client side check with the form and then change if($_POST["g-recaptcha-response"]) to if($response).

File: lib/Flux/SessionData.php (hunk 2 above)
Comment (Lavenblade) [inline lib/Flux/SessionData.php]: This is the same as above but for the login captcha.

File: lib/Flux/LoginServer.php (hunk 1 above)
Comment (MishimaHaruna) [inline lib/Flux/LoginServer.php]: Please do, it makes the purpose much more clear

**Comment by Lavenblade** [comment]
Alright, done with the changes.

Any advice on how to avoid the commit spam? I tried squashing them using my git client but they still show up on github.

**Comment by MishimaHaruna** [comment]
To squash, I do the following. It assumes that you have two remotes: `upstream` pointing to `HerculesWS/FluxCP` and `origin` pointing to `Lavenblade/FluxCP`, and that you are currently on the branch you opened the pull request from (so change as needed for your setup).

```
git fetch --all                 # Make sure you're up to date
git rebase -i upstream/master   # Rebase your branch on top of master in interactive mode
# Now change all the 'pick' into 'squash' for all the commits you want to squash;
#    leave 'pick for the first commit you want to keep, then save and quit the editor
git push --force                # Push your changes, overwriting the current content of
                                # your branch (git needs this to know that you know what
                                # you're doing, since a rebase operation is potentially
                                # dangerous, since it rewrites history, deleting and
                                # replacing existing commits from that branch)
```

But don't worry about it now, I can just squash them from the GitHub UI while merging.

For the newline issue: yeah, it's mostly the FluxCP codebase's fault though. The original author had their text editor severely misconfigured, and was leaving a mess of trailing whitespace on empty lines.

**Comment by Lavenblade** [comment]
Awesome, thanks for the tip!

Are there any other changes that need made? Would you like me to host a copy of FluxCP with this working before pulling?

**Pull request merged**